controllers/vmware/virtualmachinegroup_reconciler_test.go

```diff
@@ -0,0 +1,1227 @@
+	}{
+		// Compute new VirtualMachineGroup (start initial placement)
+		{
+			name: "compute new VirtualMachineGroup",
+			mds: []clusterv1.MachineDeployment{
+				*createMD("md1", cluster.Name, "", 2),
+				*createMD("md2", cluster.Name, "", 1),
+				*createMD("md3", cluster.Name, "zone1", 1),
+			},
+			vSphereMachines: []vmwarev1.VSphereMachine{
+				*createVSphereMachine("m1", cluster.Name, "md1", "", withCustomNamingStrategy()),
+				*createVSphereMachine("m2", cluster.Name, "md1", "", withCustomNamingStrategy()),
+				*createVSphereMachine("m3", cluster.Name, "md2", "", withCustomNamingStrategy()),
+				*createVSphereMachine("m4", cluster.Name, "md3", "zone1"),
+			},
+			existingVMG: nil,
+			want: &vmoprv1.VirtualMachineGroup{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: cluster.Namespace,
+					Name:      cluster.Name,
+					Labels: map[string]string{
+						clusterv1.ClusterNameLabel: cluster.Name,
+					},
+					Annotations: map[string]string{
+						fmt.Sprintf("%s/%s", vmoperator.ZoneAnnotationPrefix, "md3"): "zone1", // failureDomain for md3 is explicitly set by the user
+					},
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: clusterv1.GroupVersion.String(),
+							Kind:       "Cluster",
+							Name:       cluster.Name,
+							UID:        cluster.UID,
+							Controller: ptr.To(true),
+						},
+					},
+				},
+				Spec: vmoprv1.VirtualMachineGroupSpec{
+					BootOrder: []vmoprv1.VirtualMachineGroupBootOrderGroup{
+						{
+							Members: []vmoprv1.GroupMember{
+								{Name: "m1-vm", Kind: "VirtualMachine"},
+								{Name: "m2-vm", Kind: "VirtualMachine"},
+								{Name: "m3-vm", Kind: "VirtualMachine"},
+								{Name: "m4", Kind: "VirtualMachine"},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		// Compute updated VirtualMachineGroup (during initial placement)
+		{
+			name: "compute updated VirtualMachineGroup during initial placement",
+			mds: []clusterv1.MachineDeployment{
+				*createMD("md1", cluster.Name, "", 2),
+				*createMD("md3", cluster.Name, "zone1", 2),
+				*createMD("md4", cluster.Name, "zone2", 1),
+			},
+			vSphereMachines: []vmwarev1.VSphereMachine{
+				*createVSphereMachine("m1", cluster.Name, "md1", "", withCustomNamingStrategy()),
+				*createVSphereMachine("m5", cluster.Name, "md1", "", withCustomNamingStrategy()),
+				*createVSphereMachine("m4", cluster.Name, "md3", "zone1"),
+				*createVSphereMachine("m6", cluster.Name, "md3", "zone1"),
+				*createVSphereMachine("m7", cluster.Name, "md4", "zone2"),
+			},
+			existingVMG: &vmoprv1.VirtualMachineGroup{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: cluster.Namespace,
+					Name:      cluster.Name,
+					UID:       types.UID("uid"),
+					Labels: map[string]string{
+						clusterv1.ClusterNameLabel: cluster.Name,
+					},
+					Annotations: map[string]string{
+						fmt.Sprintf("%s/%s", vmoperator.ZoneAnnotationPrefix, "md3"): "zone1", // failureDomain for md3 is explicitly set by the user
```

Comment:
Let's add an annotation without the ZoneAnnotationPrefix and check it is preserved

Reply:
Updated.